### PR TITLE
Add comprehensive edge case tests for event scheduling

### PR DESCRIPTION
**Pull request by Jayantparashar10**
## Description

- Add 27 edge case tests for experimental event scheduling
- Test priority ordering, nested events, and cancellation patterns
- Test scalability (10k events), float precision, and heap property
- Test EventGenerator edge cases and weakref cleanup
- Test hybrid ABM-DEVS execution and agent interactions
- Document validation gap: negative intervals not validated

All tests pass (26/27, 1 skipped documenting validation issue).

### Summary
Adds comprehensive edge case tests for the event scheduling system (mesa.experimental.devs). Tests cover edge cases, stress scenarios, and error handling that weren't previously covered in test_devs.py.

### Motive
After PR #3204 unified the event list architecture, the event scheduling system needed more thorough edge case coverage. These tests ensure the system handles unusual scenarios correctly and provides regression protection for future changes to experimental features.

### Implementation
Created test_event_scheduling.py with three test classes:
- TestEventSchedulingEdgeCases: 23 tests covering priority ordering, nested scheduling, fractional times, generator edge cases, weakref cleanup, and error handling
- TestEventSchedulingWithRealAgents: 2 tests for agent-event interactions
- TestEventSchedulingPerformance: 2 tests verifying heap property and peek_ahead performance

Tests use named functions instead of lambdas to avoid weakref garbage collection issues. One test is intentionally skipped to document that negative intervals aren't currently validated.

### Usage Examples
Tests run with standard pytest:
```bash
pytest tests/experimental/test_event_scheduling.py -v
```

No API changes - purely additive test coverage.

## Test Results
```
26 passed, 1 skipped in 0.72s
```

### Additional Notes
- No overlap with existing test_devs.py tests
- One skipped test documents potential improvement area (negative interval validation)

### CODE REVIEW

File: tests/experimental/test_event_scheduling.py [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+        simulator.setup(model)
+
+        fn = MagicMock()
+        gen = EventGenerator(model, fn, interval=10.0)
+        gen.start(at=5.0)
+        gen.stop(at=3.0)
+
+        simulator.run_until(10.0)
+
+        assert fn.call_count >= 0
```
[the remark this answers is not in the document]
Comment (Jayantparashar10) [reply tests/experimental/test_event_scheduling.py]: Fixed! Changed to `assert fn.call_count == 1` since the first scheduled event executes even when stop is called before start time.

File: tests/experimental/test_event_scheduling.py [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+        for i in range(10000):
+            event = SimulationEvent(i, fn, priority=Priority.DEFAULT)
+            event_list.add_event(event)
+
+        start = time.time()
+        peeked = event_list.peek_ahead(100)
+        duration = time.time() - start
+
+        assert len(peeked) == 100
+        assert duration < 0.1
```
[the remark this answers is not in the document]
Comment (Jayantparashar10) [reply tests/experimental/test_event_scheduling.py:576]: This is intentional. The test is in `TestEventSchedulingPerformance` class and specifically tests peek_ahead performance. The timing assertion ensures the operation completes efficiently.

File: tests/experimental/test_event_scheduling.py [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+
+        simulator = DEVSimulator()
+        simulator.setup(model)
+
+        try:
+            gen.start()
+            pytest.skip("Negative intervals not validated - potential improvement area")
+        except ValueError:
+            pass
+
```
[the remark this answers is not in the document]
Comment (Jayantparashar10) [reply tests/experimental/test_event_scheduling.py:178]: The current logic is correct for documenting missing validation. The test tries to start the generator:
- If NO error is raised -> validation is missing -> skip to document the gap
- If ValueError is raised -> validation exists -> test passes

File: tests/experimental/test_event_scheduling.py [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+        gen = EventGenerator(model, MagicMock(), interval=-1.0)
+
+        simulator = DEVSimulator()
+        simulator.setup(model)
+
+        try:
+            gen.start()
+            pytest.skip("Negative intervals not validated - potential improvement area")
+        except ValueError:
+            pass
```
[the remark this answers is not in the document]
Comment (Jayantparashar10) [reply tests/experimental/test_event_scheduling.py:177]: The current code is clearer: if ValueError is raised (meaning validation exists), the test passes silently.

**Comment by EwoutH** [comment]
I'm open to more tests, but the event scheduling API is still in active development, so I prefer to do this stuff when we're done.

I think it also should be a bit more targeted, I don't believe all 27 of these edge cases are as relevant or likely. The fact that they all pass without any modifications is nice though (and maybe also says something).

Closing for now, but let's revisit this once the API is stable.

**Comment by Jayantparashar10** [comment]
Thanks for the feedback, @EwoutH 

You're right - 27 test scenarios is overkill, especially for an API that's still evolving. I should have focused on the 5-10 edge cases users are most likely to hit in practice.

I appreciate the lesson about waiting for API stability and being more strategic. I understand the PR closure - it makes sense not to add extensive tests before the API stabilizes.
